sno/init.py

```diff
@@ -162,7 +162,7 @@ def convert(self, value, param, ctx):
     "--similarity-detection-limit",
     hidden=True,
     type=click.INT,
-    default=50,
+    default=500,
```

Comment:
since this feature is now linear-time, I'd suggest a much higher limit like 10K or even 100K. assuming 10KB of memory per feature (fairly large features) then 100K features is only 1GB of memory, which seems passable (and in low-memory environments the user can override this flag)